Framework/API/src/NearestNeighbours.cpp

```diff
@@ -141,18 +126,16 @@ void NearestNeighbours::build(const int noNeighbours) {
   BoundingBox bbox;
   // Base the scaling on the first detector, should be adequate but we can look
   // at this
-  IDetector_const_sptr firstDet = (*spectraDets.begin()).second;
-  firstDet->getBoundingBox(bbox);
+  const auto &firstDet = m_spectrumInfo.detector(indices.front());
```

Comment:
This whole first detector stuff would be a lot nicer if it was hidden and collocated somehow . i.e `m_spectrumInfo.getRepresentiviveDetector(indexes)` otherwise the client has to know too much and these rules are too scattered.